### PR TITLE
Implement bot-user support

### PR DESCRIPTION
**Pull request by mtyurt**
I have added a checkbox which indicates given token belongs to a bot user or not.

But why?

Recently, I have figured out that if I put @username to channel parameter, the plugin sends a direct message via @slackbot. But I didn't want @slackbot to send me a dm, a bot (preferably named jenkins-bot) should send me a dm. So here it is.
- In global config and advanced job config, there is a checkbox named "Is Bot User?" (false by default). 
- If this checkbox is checked, the plugin will send a post request to https://slack.com/api/chat.postMessage .
- Also, I added all parameters to request URI if "bot-user" is true, leaving the original case as it is, because sending parameters in JSON body didn't work for chat.postMessage endpoint, I assume there is a special case for "/services/hooks/jenkins-ci". 
- Lastly, there are a few string equals improvements that I couldn't resist.

### CODE REVIEW

File: src/main/java/jenkins/plugins/slack/SlackNotifier.java [hunk 1]
```diff
@@ -66,8 +67,12 @@ public String getAuthToken() {
         return authToken;
     }
 
+    public boolean getBotUser() {
+        return botUser;
+    }
+
     public String getBuildServerUrl() {
-        if(buildServerUrl == null || buildServerUrl == "") {
+        if(buildServerUrl == null || "".equals(buildServerUrl)) {
```
Comment (rborer) [inline src/main/java/jenkins/plugins/slack/SlackNotifier.java]: You may even be tempted to use `StringUtils.isBlank(buildServerUrl)` (https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html#isBlank(java.lang.String)) :-)

**Comment by kmadel** [comment]
Any updates to my comments? Also there are a number of conflicts.

**Comment by mtyurt** [comment]
@kmadel merge conflicts are now resolved. I'm confused about the comments, am I the target of these comments? Because I did not get any notification about it.

**Comment by kmadel** [comment]
@mtyurt Not sure why you didn't get notified. But yes, those comments were directed at you. Need 'Is Bot' specific help file. Also, please update the README.md to document this feature. My plan it move all documentation into README and direct users from the Jenkins Wiki to the README. And Thanks!

**Comment by mtyurt** [comment]
@kmadel from your comment I understand all documentation should be in README file, however there aren't any documentation regarding configurations in README so far. I'll change the screenshot including bot user option, but including configuration info to README should be subject to another pull request.

**Comment by kmadel** [comment]
@mtyurt Yes, please add `Bot User Support` section to README as you mentioned - I know the README is still very much lacking, but we will get it where it needs to be one merge at a time.
Thanks!

Also, would be nice to get a few more thumbs up from a code review perspective.

**Comment by mtyurt** [comment]
@kmadel it's done, can you guys have a look please?

**Comment by samrocketman** [comment]
Feel free to subscribe to #296 for updates related to Slack Plugin 2.2 release.

**Pull request merged**